File: parse_models.py
import json

with open('models.json', 'rb') as f:
    content = f.read().decode('utf-8-sig')  # utf-8-sig handles BOM
    data = json.loads(content)

free_models = []
for model in data['data']:
    pricing = model['pricing']
    if all(float(pricing.get(k, 1)) == 0 for k in ['prompt', 'completion', 'request', 'image', 'web_search', 'internal_reasoning']):
        free_models.append({
            'id': model['id'],
            'name': model['name']
        })

print(json.dumps(free_models, indent=2))

File: parse_models.js
const fs = require("fs");

let content = fs.readFileSync("models.json", "utf8");
if (content.charCodeAt(0) === 0xfeff) {
  content = content.slice(1);
}
const data = JSON.parse(content);

const freeModels = [];
for (const model of data.data) {
  const pricing = model.pricing;
  if (
    parseFloat(pricing.prompt) === 0 &&
    parseFloat(pricing.completion) === 0 &&
    parseFloat(pricing.request) === 0 &&
    parseFloat(pricing.image) === 0 &&
    parseFloat(pricing.web_search) === 0 &&
    parseFloat(pricing.internal_reasoning) === 0
  ) {
    freeModels.push({
      id: model.id,
      name: model.name,
    });
  }
}

console.log(JSON.stringify(freeModels, null, 2));
